src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -616,10 +616,19 @@ private List<LaunchCommand> GetInitializeCommands()
 
                     CheckCygwin(commands, localLaunchOptions);
 
-                    // ClrDbg doesn't need it.
-                    if (this.MICommandFactory.Mode != MIMode.Clrdbg)
+                    // ClrDbg doesn't need -file-exec-and-symbols set.
+                    if (this.MICommandFactory.Mode == MIMode.Gdb)
                     {
-                        this.AddExecutablePathCommand(commands);
+                        if (_launchOptions is UnixShellPortLaunchOptions)
```

Comment:
>_launchOptions is UnixShellPortLaunchOptions [](start = 28, length = 44)

Since your code doesn't actually depend on any particular transport, would it make sense to instead check if the executable path is set or not defined?